### ISSUE TITLE
bug: press % for matching does not work or goes to wrong locations

### ISSUE THREAD

**Issue by Viperform** [VSCodeVim/Vim#651]
Please *thumbs-up* 👍 this issue if it personally affects you! You can do this by clicking on the emoji-face on the top right of this post. Issues with more thumbs-up will be prioritized.

-----

### What did you do?

Open this file https://github.com/telerik/kendo-ui-core/blob/master/src/kendo.color.js  in visual studio code and switch to NORMAL MODE, go to last line which begins with "}, typeof define == 'function' && define.amd", move cursor to col 1, that's should be under '}', now press %.

### What did you expect to happen?
The cursor should go to '{' in line 3.

### What happened instead?
Nothing happens.
Many times, press % will cause the cursor go to wrong places.

### Technical details:

* VSCode Version: 1.3 /1.4
* VsCodeVim Version: *[please ensure you are on the latest]* 0.1.7
* OS: win7, 64bit

**Comment by johnfn** [comment]
Yeah, this is a known issue unfortunately, We don't correctly handle cases like `( "(" )`. Feel free to add it if you want 😉

**Comment by xconverge** [comment]
@johnfn This seems to be fixed?

**Comment by jpoon** [comment]
Indeed, this looks to be working on 0.2.0.

**Comment by johnfn** [comment]
That's super weird, because our matching algorithm has never changed!

https://github.com/VSCodeVim/Vim/blob/master/src/matching/matcher.ts#L21

Something like `( ")" )` continues to trip us up, so I'm gonna keep this one open for now. 😉

**Comment by Chillee** [comment]
Is there any reason we don't just use`editor.action.jumpToBracket`?

**Comment by Chillee** [comment]
Oh, here's one... It doesn't support multiple cursors >:(

**Comment by johnfn** [comment]
Ah multi cursors, the cause of, and solution to, all of VSCodeVim's problems.

**Comment by antonburger-advanced** [comment]
Also affects motions like <kbd>[</kbd><kbd>{</kbd> and <kbd>]</kbd><kbd>}</kbd>. Would guess the same code is involved in finding pairs here?

**Comment by bijoutrouvaille** [comment]
I don't use multiple cursors, so I made this mapping and it has worked so far:

```
{
      "before": ["%"],
      "commands": [
        {
          "command": "editor.action.jumpToBracket",
          "args": []
        },
        {
          "command": "editor.action.moveCarretLeftAction",
          "args": []
        },
        {
          "command": "editor.action.moveCarretRightAction",
          "args": []
        }
      ]
    }
```
The left-right carret motion at the end makes the window scroll to the new carret position.

**Comment by dossy** [comment]
The use of `editor.action.moveCarretLeftAction` and `editor.action.moveCarretRightAction` was the trick I needed to get `%` to work properly.  I had found `editor.action.jumpToBracket` and bound it to `%` in normal mode, but it would only pretend to move the cursor to the matching bracket; any kind of interaction in normal mode would "jump" back to the previous cursor location.

Thank you, @bijoutrouvaille!  My prefs JSON snippet looks like this:

```
    "vim.normalModeKeyBindingsNonRecursive": [
        {
            "before": [
                "%"
            ],
            "commands": [
                "editor.action.jumpToBracket",
                "editor.action.moveCarretLeftAction",
                "editor.action.moveCarretRightAction"
            ]
        }
    ]
```

**Comment by drdo** [comment]
Why can't `%` just be bound to `editor.action.jumpToBracket`? Seems to work for me.

By the way this is also broken in the same way for `va[` and similar commands.
